src/containers/wildcam-classrooms/WildCamClassroomFormContainer.jsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Actions } from 'jumpstate';
+
+import WildCamClassroomForm from '../../components/wildcam-classrooms/WildCamClassroomForm';
+import {
+  CLASSROOMS_INITIAL_STATE, CLASSROOMS_PROPTYPES
+} from '../../ducks/classrooms';
+import {
+  PROGRAMS_INITIAL_STATE, PROGRAMS_PROPTYPES
+} from '../../ducks/programs';
+
+export class WildCamClassroomFormContainer extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.autoCreateAssignments = this.autoCreateAssignments.bind(this);
+    this.createClassroom = this.createClassroom.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.closeForm = this.closeForm.bind(this);
+    this.updateClassroom = this.updateClassroom.bind(this);
+  }
+
+  componentWillUnmount() {
+    this.resetFormFields();
+  }
+
+  onChange(event) {
+    const fields = { ...this.props.formFields, [event.target.id]: event.target.value };
+    Actions.classrooms.updateFormFields(fields);
+  }
+
+  onSubmit(event) {
+    event.preventDefault();
+    if (this.props.selectedClassroom) {
+      this.updateClassroom().then(() => {
+        Actions.getClassroom(this.props.selectedClassroom.id);
+        this.closeForm();
+      });
+    } else {
+      this.createClassroom().then(() => {
+        Actions.getClassroomsAndAssignments(this.props.selectedProgram);
+        this.closeForm();
+      });
+    }
+  }
+
+  resetFormFields() {
+    Actions.classrooms.updateFormFields(CLASSROOMS_INITIAL_STATE.formFields);
+  }
+
+  closeForm() {
+    Actions.classrooms.toggleFormVisibility();
+  }
+
+  createClassroom() {
+    const classroomData = {
+      attributes: this.props.formFields,
+      relationships: {
+        program: {
+          data: {
+            id: this.props.selectedProgram.id,
+            type: 'programs'
+          }
+        }
+      }
+    };
+
+    return Actions.createClassroom(classroomData)
+      .then((classroom) => {
+        if (!this.props.selectedProgram.custom) {
```

Comment:
This only applies to I2A (a program with `custom: true` set). This is non-blocking comment and in my overall PR review, I'll go into the shared API request needs more.